### PR TITLE
Update service name to allow including . - _ chars.

### PR DESCRIPTION
**Pull request by rupakg**
[Finishes #101076518]

See UI equivalent rule at https://github.com/CenturyLinkLabs/lorry-ui/pull/167

### CODE REVIEW

File: app/models/compose_validator.rb [hunk 1]
```diff
@@ -100,16 +100,16 @@ def validate_expose_format(value, path, errors)
...
         service_name = path[0]
         if value.include?('build')
           unless service_name =~ regex_with_build
-            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-z0-9]', path)
+            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-z0-9._-] but not starting with [.-_]', path)
           end
         else
           unless service_name =~ regex
-            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-zA-Z0-9]', path)
+            errors << Kwalify::ValidationError.new('Invalid service name. Valid characters are [a-zA-Z0-9._-] but not starting with [.-_]', path)
```
Comment (alexwelch) [inline app/models/compose_validator.rb]: Is this true (that you can't begin with one of these characters)? They don't really mention it here: https://github.com/docker/compose/blob/fb4c9fbb8ba2797284296601eb904dd24c6a7b47/compose/config/validation.py#L27, but I assume you actually tried it...

**Comment by alexwelch** [comment]
:+1: if you've addressed the above comment.

File: app/models/compose_validator.rb (hunk 1 above)
Comment (rupakg) [inline app/models/compose_validator.rb]: @alexwelch For a service name that has a 'build' component to it, docker-compose throws this error:
```
Building _custom_code-ver.1...
repository name component must match "[a-z0-9]+(?:[._-][a-z0-9]+)*"
```
But for a service name that does not use 'build', I noticed that docker-compose did not throw a validation error, but it did choke on the name, and the service would not start. See the output below:
```
Creating app2_custom_code-ver.1_1...
Creating app2__Web_service-ver.2_1...
Attaching to app2_custom_code-ver.1_1, app2__Web_service-ver.2_1
^CGracefully stopping... (press Ctrl+C again to force)
Stopping app2__Web_service-ver.2_1... done
```
So, I decided to disallow a service name to start with [.-_] to be consistent and not be buggy. 

What are your thoughts?

Comment (alexwelch) [inline app/models/compose_validator.rb]: I think you did the right thing

**Pull request merged**